py/src/nupic/examples/hotgym.py

```diff
@@ -0,0 +1,161 @@
+    synPermConnected           = spParams["synPermConnected"],
+    boostStrength              = spParams["boostStrength"],
+    wrapAround                 = True
+  )
+  sp_info = Metrics( sp.getColumnDimensions(), 999999999 )
+
+  tmParams = parameters["tm"]
+  tm = TemporalMemory(
+    columnDimensions          = (spParams["columnCount"],),
+    cellsPerColumn            = tmParams["cellsPerColumn"],
+    activationThreshold       = tmParams["activationThreshold"],
+    initialPermanence         = tmParams["initialPerm"],
+    connectedPermanence       = spParams["synPermConnected"],
+    minThreshold              = tmParams["minThreshold"],
+    maxNewSynapseCount        = tmParams["newSynapseCount"],
+    permanenceIncrement       = tmParams["permanenceInc"],
+    permanenceDecrement       = tmParams["permanenceDec"],
+    predictedSegmentDecrement = 0.0,
+    maxSegmentsPerCell        = tmParams["maxSegmentsPerCell"],
+    maxSynapsesPerSegment     = tmParams["maxSynapsesPerSegment"]
+  )
+  tm_info = Metrics( [tm.numberOfCells()], 999999999 )
+
+  anomaly_history = AnomalyLikelihood()
+
+  # Read the input file.
+  records = []
+  with open(_INPUT_FILE_PATH, "r") as fin:
+    reader = csv.reader(fin)
+    headers = next(reader)
+    next(reader)
+    next(reader)
+    for record in reader:
+      records.append(record)
+  for i in range(100):
+    records.append( random.choice(records) )
+
+  inputs  = []
+  anomaly = []
+  for record in records:
+
+    # Convert data string into Python date object.
+    dateString = datetime.datetime.strptime(record[0], "%m/%d/%y %H:%M")
+    # Convert data value string into float.
+    consumption = float(record[1])
+
+    # Now we call the encoders to create bit representations for each value.
+    timeOfDayBits   = timeOfDayEncoder.encode(dateString)
+    weekendBits     = weekendEncoder.encode(dateString)
+    consumptionBits = scalarEncoder.encode(consumption)
+
+    # Concatenate all these encodings into one large encoding for Spatial
+    # Pooling.
+    encoding = SDR( encodingWidth ).concatenate(
+                                [consumptionBits, timeOfDayBits, weekendBits])
+    enc_info.addData( encoding )
+
+    # Create an SDR to represent active columns, This will be populated by the
+    # compute method below. It must have the same dimensions as the Spatial
+    # Pooler.
+    activeColumns = SDR( sp.getColumnDimensions() )
+
+    # Execute Spatial Pooling algorithm over input space.
+    sp.compute(encoding, True, activeColumns)
+    sp_info.addData( activeColumns )
+
+    # Execute Temporal Memory algorithm over active mini-columns.
+    tm.compute(activeColumns, learn=True)
+    tm_info.addData( tm.getActiveCells().flatten() )
+
+    anomalyLikelihood = anomaly_history.anomalyProbability( consumption, tm.anomaly )
+
+    inputs.append( consumption )
+    anomaly.append( tm.anomaly )
+    # anomaly.append( anomalyLikelihood )
```

Comment:
also plot anomaly? Ideally not adding it to anomaly, but keeping the two separate